samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -24,8 +24,11 @@
 import org.apache.samza.operators.triggers.Trigger;
 import org.apache.samza.operators.triggers.Triggers;
 import org.apache.samza.operators.windows.internal.WindowInternal;
+import org.apache.samza.operators.windows.internal.WindowType;
 
 import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.function.BiFunction;
 import java.util.function.Function;
```

Comment:
Good point. Here's the other side of your question -  If we take the "early trigger" as a parameter to the `globalwindow`, and if a user chooses to `setEarlyTrigger` on it, what the behavior should be? (we can choose to obviously clobber the configured earlyTrigger)

For the MVP, we agreed that `globalwindow` is not supported anyways. I'll remove it from the API. We can consider adding it later if there's a strong use-case.